subgraphs/nft-market/mappings/utils/dayUpdates.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable prefer-const */
+import { CollectionDayData, MarketPlaceDayData } from "../../generated/schema";
+import { Address, BigInt, BigDecimal, ethereum } from "@graphprotocol/graph-ts";
+
+// BigNumber-like references
+let ONE_BI = BigInt.fromI32(1);
+
+export function updateCollectionDayData(
+  collection: Address,
+  newVolumeinBNB: BigDecimal,
+  event: ethereum.Event
+): CollectionDayData {
+  let timestamp = event.block.timestamp.toI32();
+  let dayID = timestamp / 86400;
+  let dayStartTimestamp = dayID * 86400;
+
+  let ID = dayID.toString() + "-" + collection.toHexString();
+
+  let collectionDayData = CollectionDayData.load(ID.toString());
+
+  if (collectionDayData === null) {
+    collectionDayData = new CollectionDayData(ID.toString());
+    collectionDayData.date = dayStartTimestamp;
+    collectionDayData.collection = collection.toHexString();
+    collectionDayData.dailyVolumeBNB = newVolumeinBNB;
+    collectionDayData.dailyTrades = ONE_BI;
+  } else {
+    collectionDayData.dailyVolumeBNB = collectionDayData.dailyVolumeBNB.plus(newVolumeinBNB);
+    collectionDayData.dailyTrades = collectionDayData.dailyTrades.plus(ONE_BI);
+  }
+
+  collectionDayData.save();
+  return collectionDayData as CollectionDayData;
```

Comment:
You don't need to return the entity since it's not assigned anywhere in your main mapping